### DISCUSSION TITLE
Generating a password for the text push

### DISCUSSION THREAD

**Discussion by CBB3810** [pglombardo/PasswordPusher#2415] [Q&A] [answered]
I researched https://pwpush.com/api/1.0/passwords.en.html
I could not find an api call that actually generates a password *and* puts that password into the text push API call. Is there such a thing?

**Comment by CBB3810** [comment]
I could easily run a script that would let me do that with a separate linux command like: openssl rand -base64 16
But, since there's a generate password button, I have to assume there is functionality built into the PasswordPusher software to do so?

**Comment by pglombardo** [comment, marked as answer]
That's correct that it's not there.  The password generator hasn't made it into the API (yet).

**Reply by CBB3810** [reply]
I made a temp workaround script and could add a postfix routine if I wanted to:
```
#!/bin/bash

# Generate a random password with special characters
password=$(openssl rand -base64  18 | tr -d '+/')

# Make the API call with the generated password
curl -X POST --data "password[payload]=$password&password[expire_after_days]=7&password[expire_after_views]=1" http://your.site.local:5100/p.json | jq -r '.url_token' | xargs -I {} echo "http://your.site.local:5100/en/p/{}/r"

# Output the generated password for reference (optional)
echo "Generated password: $password"
```

**Reply by CBB3810** [reply]
The portal doesn't display '+" symbols when viewing the password.

**Reply by pglombardo** [reply]
It seems to work for me: https://pwpush.com/p/j3sbi_zcfh26pmwp

Might it be a unicode character possibly?  Could you post here a string that doesn't display?

**Comment by pglombardo** [comment]
A new password generator will be going out in the next release.  It has been merged to master and will go through final testing.

Pre-release is live here if you want to try it out: https://oss.pwpush.com/

This release will also include API based password generation: https://oss.pwpush.com/help/api

This will be released for OSS first.  Pro/pwpush.com next.

Thanks @CBB3810  for the feedback & patience!

<img width="695" height="721" alt="Image" src="https://github.com/user-attachments/assets/7c8a59e5-f52e-469f-9d7e-fd2f67f5c34e" />
